thirdparty/nginx/nginx-1.7.1-zimbra/src/mail/ngx_mail_proxy_module.c

```diff
@@ -1763,6 +1770,13 @@ ngx_mail_proxy_internal_server_error(ngx_mail_session_t *s)
         ngx_log_debug1(NGX_LOG_DEBUG_MAIL, s->connection->log, 0,
                        "close mail proxy connection: %d",
                        s->proxy->upstream.connection->fd);
+#if (NGX_MAIL_SSL)
+
+        if (s->proxy->upstream.connection->ssl) {
+            s->proxy->upstream.connection->ssl->no_wait_shutdown = 1;
+            ngx_ssl_shutdown(s->proxy->upstream.connection);
```

Comment:
Appropriate action should be taken based on the return value of ngx_ssl_shutdown.